Common/Orders/StopMarketOrder.cs

```diff
@@ -36,6 +36,62 @@ public override decimal Value
         }
 
         /// <summary>
+        /// Create update request for pending orders. Null values will be ignored.
+        /// </summary>
+        public UpdateOrderRequest UpdateRequest(int? quantity = null, decimal? stopPrice = null, string tag = null)
+        {
+            return new UpdateOrderRequest
+            {
+                Id = Guid.NewGuid(),
+                OrderId = Id,
+                Created = DateTime.Now,
+                Quantity = quantity ?? Quantity,
+                StopPrice = stopPrice ?? StopPrice,
+                Tag = tag ?? Tag
+            };
+        }
+
+        /// <summary>
+        /// Apply changes after the update request is processed.
+        /// </summary>
+        public override void ApplyUpdate(UpdateOrderRequest request)
+        {
+            base.ApplyUpdate(request);
+
+            StopPrice = request.StopPrice;
+        }
+
+        /// <summary>
+        /// Create submit request.
+        /// </summary>
+        public static SubmitOrderRequest SubmitRequest(string symbol, int quantity, decimal stopPrice, string tag, SecurityType securityType, decimal price = 0, DateTime? time = null)
```

Comment:
Please make the time parameter required. This needs to be set as the algorithm time and I would recommend a new signature as follows:
```C#
SubmitOrderRequest CreateSubmitRequest(SecurityType securityType, string symbol, int quantity, DateTime time, decimal stopPrice, string tag = null)
```